monitor/internal/pod/delete_controller.go

```diff
@@ -14,7 +16,7 @@ import (
 )
 
 // deleteControllerReconcileFunc is the reconciler function signature for the DeleteController
-type deleteControllerReconcileFunc func(context.Context, client.Client, *config.ProcessorConfig, time.Duration, *map[string]client.ObjectKey)
+type deleteControllerReconcileFunc func(context.Context, client.Client, *config.ProcessorConfig, time.Duration, map[string]DeleteObject, extractors.PodSandboxExtractor)
```

Comment:
this is a pointer on purpose and must be a pointer: we don't want to pass copies of this map

Reply:
I thought in go the maps are passed as a pointer/by reference by default. 
The compiler usually takes care of passing the reference to the next function.